airflow/providers/amazon/aws/transfers/s3_to_dynamodb.py

```diff
@@ -0,0 +1,256 @@
+
+    @property
+    def tmp_table_name(self):
+        """Temporary table name."""
+        return f"{self.dynamodb_tmp_table_prefix}_{self.dynamodb_table_name}"
+
+    def _load_into_new_table(self, context: Context, tmp_table: bool = False) -> str:
+        """
+        Import S3 key or keys into a new DynamoDB table.
+
+        :param context: the current context of the task instance
+        :param tmp_table: creates DynamoDB Table with tmp prefix
+        :return: the Amazon resource number (ARN)
+        """
+        dynamodb_hook = DynamoDBHook(aws_conn_id=self.aws_conn_id)
+        client = dynamodb_hook.client
+        table_name = self.tmp_table_name if tmp_table else self.dynamodb_table_name
+
+        import_table_config = self.import_table_kwargs or {}
+        import_table_creation_config = self.import_table_creation_kwargs or {}
+
+        try:
+            response = client.import_table(
+                S3BucketSource={
+                    "S3Bucket": self.s3_bucket,
+                    "S3KeyPrefix": self.s3_key,
+                },
+                InputFormat=self.input_format,
+                TableCreationParameters={
+                    "TableName": table_name,
+                    "AttributeDefinitions": self.dynamodb_attributes,
+                    "KeySchema": self.dynamodb_key_schema,
+                    "BillingMode": self.billing_mode,
+                    **import_table_creation_config,
+                },
+                **import_table_config,
+            )
+        except ClientError as e:
+            self.log.error("Error: failed to load from S3 into DynamoDB table. Error: %s", str(e))
+            raise AirflowException(f"S3 load into DynamoDB table failed with error: {e}")
+
+        if response["ImportTableDescription"]["ImportStatus"] == "FAILED":
+            raise AirflowException(
+                "S3 into Dynamodb job creation failed. Code: "
+                f"{response['ImportTableDescription']['FailureCode']}. "
+                f"Failure: {response['ImportTableDescription']['FailureMessage']}"
+            )
+
+        if self.wait_for_completion:
+            self.log.info("Waiting for S3 into Dynamodb job to complete")
+            waiter = dynamodb_hook.get_waiter("import_table")
+            try:
+                waiter.wait(
+                    ImportArn=response["ImportTableDescription"]["ImportArn"],
+                    WaiterConfig={"Delay": self.check_interval, "MaxAttempts": self.max_attempts},
+                )
+            except WaiterError:
+                status, error_code, error_msg = dynamodb_hook.get_import_status(
+                    response["ImportTableDescription"]["ImportArn"]
+                )
+                client.delete_table(TableName=table_name)
+                raise AirflowException(
+                    f"S3 import into Dynamodb job failed: Status: {status}. Error: {error_code}. Error message: {error_msg}"
+                )
```

Comment:
Got it. Thanks for the explanation. In fact, I implemented it so that in case of problems running the import (network or server side related), users might want to add a simple task retry. If we don't cleanup, retrying will result in another new error, because the table already exist. If that makes sense, I would change it to an optional flag `delete_on_error` with the default value set to False. I also added a unittest for the behavior of `delete_on_error`.

Reply:
I like that, thanks!